client/core/getResource.go

```diff
@@ -31,7 +31,7 @@ func (d *Device) GetResource(
 	link schema.ResourceLink,
 	response interface{},
 	options ...coap.OptionFunc,
-) error {
+) (coap.ResponseInfo, error) {
```

Comment:
I m not sure with this. Via codec you can this informations, so I don't want add another return value